### DISCUSSION TITLE
A array map displays the rank data, starting from the third item, the dynamic w- attribute is not added to the div

### DISCUSSION THREAD

**Discussion by yqz945** [tailwindlabs/tailwindcss#11568] [Help]
**What version of Tailwind CSS are you using?**

"tailwindcss": "^3.3.2",

**What build tool (or framework if it abstracts the build tool) are you using?**

 "next": "^13.4.8",

**What version of Node.js are you using?**

v18.13.0

**What browser are you using?**

Edge

**What operating system are you using?**

Win11

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**

Array map displays the rank data, starting from the second item, the dynamic w- attribute is not added to the div.
The demo data:
`
export const RankData = [
  {
    rank: 'NO.1',
    name: '济南市',
    value: 90
  },
  {
    rank: 'NO.2',
    name: '青岛市',
    value: 80
  },
  {
    rank: 'NO.3',
    name: '烟台市',
    value: 70
  },
  {
    rank: 'NO.4',
    name: '潍坊市',
    value: 60
  },
  {
    rank: 'NO.5',
    name: '临沂市',
    value: 50
  },
  {
    rank: 'NO.6',
    name: '济宁市',
    value: 40
  },
  {
    rank: 'NO.7',
    name: '淄博市',
    value: 30
  },
  {
    rank: 'NO.8',
    name: '菏泽市',
    value: 20
  },
  {
    rank: 'NO.9',
    name: '德州市',
    value: 10
  },
  {
    rank: 'NO.10',
    name: '东营市',
    value: 5
  },
]
`

And the component code :
```
import React from 'react'
import {RankData} from "@/constants/data";

type RankItemType = {
  rank: string,
  name: string,
  value: number
}
const RankItem = (item: RankItemType) => (
  <div className="grid grid-cols-8 gap-2">
    <div className="col-span-2">{item.rank}</div>
    <div className="col-span-6 grid grid-cols-8 items-center gap-1">
      <div className="col-span-2">{item.name}</div>
      <div className="ol-span-6 h-2 w-[200px] bg-gray-600">
        <div className={"h-2 w-[" + (item.value).toString() + "px] bg-amber-600 "}>&nbsp;</div>
        {/*<div className={"h-2 bg-amber-600 "} style={{width: (item.value*2).toString() + "px"}}>&nbsp;</div>*/}
      </div>
    </div>
  </div>
)
const ActivityRank = () => {
  return (
    <div className="flex flex-col gap-4">
      {RankData.map((item, index) => <RankItem key={index} name={item.name} rank={item.rank} value={item.value}/>)}
    </div>
  )
}
export default ActivityRank
```

And the first and the second div add the width property:
.w-\[90px\] {
    width: 80px;
}

The third div don't have the width property at all. I don't know why, thanks a lot.

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

For your situation, it seems like `item.value` could have a wide range of values. So consider applying the `width` in an inline style instead:

```diff
 <div className="col-span-2">{item.name}</div>
 <div className="ol-span-6 h-2 w-[200px] bg-gray-600">
-  <div className={"h-2 w-[" + (item.value).toString() + "px] bg-amber-600 "}>&nbsp;</div>
+  <div className="h-2 bg-amber-600" style={{ width: `${item.value}px` }}>&nbsp;</div>
   {/*<div className={"h-2 bg-amber-600 "} style={{width: (item.value*2).toString() + "px"}}>&nbsp;</div>*/}
 </div>
